fastplotlib/layouts/_gridplot.py

```diff
@@ -92,12 +92,35 @@ def __init__(
                 self.shape
             )
 
+        if "names" in kwargs.keys():
+            self.names = to_array(kwargs["names"])
+            if self.names.shape != self.shape:
+                raise ValueError(f"subplot names: {self.names} must be in gridplot shape: {self.shape}")
+        else:
+            self.names = None
+
         if isinstance(controllers, str):
             if controllers == "sync":
                 controllers = np.zeros(
                     self.shape[0] * self.shape[1], dtype=int
                 ).reshape(self.shape)
 
+        # Check if 'names' and 'controllers' both have values; check if values in controller array are strings
+        if (self.names is not None) and (controllers is not None):
+            c = to_array(controllers)
+            if isinstance(c[0, 0], str):
+                idx_async_controllers = np.shape(c)[0]
+                controllers = np.zeros(
+                    self.shape[0] * self.shape[1], dtype=int
+                ).reshape(self.shape)
+                for positions in product(range(self.shape[0]), range(self.shape[1])):
+                    item = np.argwhere(c == self.names[positions])
```

Comment:
I don't understand what `item` is, can you write this clearer with a more informative name?